packages/blade/src/components/FileUpload/_decisions/decisions.md

```diff
@@ -0,0 +1,373 @@
+   */
+  accessibilityLabel?: string;
+  /**
+   * Position of the label relative to the file upload area. Desktop only prop. Default value on mobile will be 'top'
+   */
+  labelPosition?: 'top' | 'left';
+  /**
+   * Defines the selection behavior of the FileUpload component
+   */
+  selectionType?: 'single' | 'multiple';
+  /**
+   * File types that can be accepted. See [input's accept attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept)
+   * 
+   * Usage: accept=".jpg, .png, .pdf", accept="image/*", accept="image/png, image/jpeg, application/pdf" 
+   */
+  accept?: string;
+  /**
+   * Disables or enables the FileUpload component
+   */
+  isDisabled?: boolean;
+  /**
+   * The name of the file upload input, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)
+   */
+  name?: string;
+  /**
+   * Default list of files that have been uploaded, useful when the component is uncontrolled
+   */
+  defaultFileList?: FileList;
+  /**
+   * List of files that have been selected/uploaded, useful when the component is controlled
+   */
+  fileList?: FileList;
+  /**
+   * Limit the number of files that can be uploaded
+   */
+  maxCount?: number;
+  /**
+   * Limit the size of the uploaded files (in bytes)
+   */
+  maxSize?: number;
+  /**
+   * Callback function triggered when files are selected
+   */
+  onChange?: ({ name, fileList }: { name: string; fileList: FileList }) => void;
+  /**
+   * Callback function triggered when the preview icon is clicked
+   */
+  onPreview?: (previewedFile: File) => void;
+  /**
+   * Callback function triggered when a file is removed
+   */
+  onRemove?: ({ removedFile, fileList }: { removedFile: File; fileList: FileList }) => void;
+  /**
+   * Callback function executed when files are dropped into the upload area
+   */
+  onDrop?: (event: React.DragEvent) => void;
+  /**
+   * State indicating whether there is an error in the FileUpload component
+   */
+  validationState?: 'none' | 'error';
+  /**
+   * Additional text providing assistance or guidance
+   */
+  helpText?: string;
+  /**
+   * Text indicating an error state
+   */
+  errorText?: string;
+};
+```
+
+## Usage
+
+Primarily, there are two types of usage for the `FileUpload` component:
+
+### Submit Button File Upload (Uncontrolled)
+
+https://github.com/razorpay/blade/assets/46647141/d78ec694-f6f7-413d-a32c-58da8f2582fb
+
+### Direct Upload with Progress Display (Controlled)
+
+https://github.com/razorpay/blade/assets/46647141/cb690625-011b-4c20-b610-9af5f0c5e077
+
+## Examples
+
+Here are a few illustrative examples showcasing the utilization of the `FileUpload` component:
+
+### Uncontrolled Usage
+
+#### Single File selection
+
+```jsx
+import React, { useState } from 'react';
+import axios from 'axios';
+import { Box, FileUpload } from '@razorpay/blade/components';
+
+const UncontrolledSingleFileUploadForm = () => {
+  const [file, setFile] = useState();
```

Comment:
How is this uncontrolled when I'm actually managing the `file` state? 

Ideally we should not have this and only have handleSubmit? 

```
handleSubmit = (e) => {
  const files = e.target.files;
  // do something
}
```

Reply:
We aren't passing it to the component, we don't need this to attach/select/drop the files. Users need this information when they upload files.

Reply:
Maybe I can update the example using ref

Reply:
Updated.